### ISSUE TITLE
Publish of DLL with RID not longer includes shared framework bits.

### ISSUE THREAD

**Issue by ericstj** [dotnet/sdk#1198]
Using the following project:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp1.1</TargetFramework>
    <RuntimeIdentifiers>win7-x64;win7-x86</RuntimeIdentifiers>
  </PropertyGroup>

</Project>
```

Run `dotnet publish -f netcoreapp1.1 -r win7-x64`

In 1.1.0 RTM you'd get a standalone layout.

In latest 2.0 preview bits you only get the DLL.

If I change outputtype to Exe or explicitly set SelfContained it gets the standalone layout.

/cc @eerhardt

**Comment by nguerrera** [comment]
@eerhardt Is there a problem with the defaulting of SelfContained here?

**Comment by eerhardt** [comment]
Yes.  The SelfContained defaulting checks for `$(HasRuntimeOutput) == true`

https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.RuntimeIdentifierInference.targets#L107-L110

It wasn't expected to have a scenario where you are publishing a non application and wanted the runtime bits published to your output folder.

**Comment by livarcocc** [comment]
@ericstj what is the scenario here? In which cases do we want this behavior?

**Comment by ericstj** [comment]
The scenario where I've seen multiple people hit this is creating a project to just publish some packages (they didn't care about the output of the project itself). In those cases folks don't tend to set outputtype because it makes no sense to do so and they just want a minimal project to publish.

**Comment by livarcocc** [comment]
But in this case, could we simply tell them to set the OutputType then, since they don't really care?

**Comment by ericstj** [comment]
Sure you could tell folks to change their projects to workaround a breaking change or you could not have the breaking change 😉

**Comment by nguerrera** [comment]
I think we should change the behavior back to the way it was in 1.x.

**Comment by nguerrera** [comment]
Well, we shipped with the breaking behavior in 2.0. :( I guess we can chalk it up to semver, but I think this was unfortunate. Not sure it's worth changing back now, though.
